precovery/frame_db.py

```diff
@@ -617,7 +651,14 @@ def store_observations(self, observations: Iterable[Observation]):
         which indicates the ID's length in bytes, followed by the UTF8-encoded
         bytes of the ID.
         """
-        f = self._current_data_file()
+        f = None
+        try:
+            f = self._current_data_file(dataset_id, year_month_str)
+        except KeyError as ke:
```

Comment:
The implicit contract here - that `self._current_data_file` returns a KeyError if the datafile doesn't exist - feels a bit brittle. It's internal enough that it might not be worth fixing, but just worth giving a second thought.

Reply:
Agreed completely. Maybe it would make more sense for `_current_data_file` to have a create=True flag, create a new datafile if necessary, and avoid the KeyError nonsense altogether.